In [1]:
import pandas as pd

all_movies = pd.read_csv("/Users/iliasrazig/Documents/ENSAE/S1/Infrastructure/Projet/all_movies_ratings.csv")

In [3]:
all_movies.head()

movieId      title    year                                       genres  \
0        1  Toy Story  1995.0  Adventure|Animation|Children|Comedy|Fantasy   
1        1  Toy Story  1995.0  Adventure|Animation|Children|Comedy|Fantasy   
2        1  Toy Story  1995.0  Adventure|Animation|Children|Comedy|Fantasy   
3        1  Toy Story  1995.0  Adventure|Animation|Children|Comedy|Fantasy   
4        1  Toy Story  1995.0  Adventure|Animation|Children|Comedy|Fantasy   

   userId  rating   timestamp  
0       2     3.5  1141415820  
1       3     4.0  1439472215  
2       4     3.0  1573944252  
3       5     4.0   858625949  
4       8     4.0   890492517

In [6]:
all_movies[all_movies['genres']=='(no genres listed)']

movieId                           title    year              genres  \
22550136    83773  Away with Words (San tiao ren)  1999.0  (no genres listed)   
22550137    83773  Away with Words (San tiao ren)  1999.0  (no genres listed)   
22575372    84768                      Glitterbug  1994.0  (no genres listed)   
22575373    84768                      Glitterbug  1994.0  (no genres listed)   
22575374    84768                      Glitterbug  1994.0  (no genres listed)   
...           ...                             ...     ...                 ...   
25000070   209103          Tsar Ivan the Terrible  1991.0  (no genres listed)   
25000076   209133          The Riot and the Dance  2018.0  (no genres listed)   
25000077   209133          The Riot and the Dance  2018.0  (no genres listed)   
25000087   209151                 Mao Zedong 1949  2019.0  (no genres listed)   
25000093   209169                    A Girl Thing  2001.0  (no genres listed)   

          userId  rating   timestamp  
22550136   26031     3.0  1299099386  
22550137   72315     0.5  1461698005  
22575372   72315     3.0  1450166864  
22575373  119765     2.5  1298874243  
22575374  137293     3.0  1298899248  
...          ...     ...         ...  
25000070   13737     4.0  1574112239  
25000076  127710     4.5  1574186235  
25000077  145795     0.5  1574225372  
25000087  125475     3.5  1574232248  
25000093  119571     3.0  1574291826  

[26627 rows x 7 columns]

10,6% des avis portent sur des films n'ayant pas de genre

In [11]:
(len(all_movies[all_movies['genres']=='(no genres listed)']['genres'])/len(all_movies['genres']))*100

0.10650759527113798

In [22]:
len(all_movies[all_movies['genres']=='(no genres listed)']['title'].unique().tolist())

4545

Ces films sont au nombre de 4545 et représentent 8,2% de l'ensemble des films

In [23]:
(len(all_movies[all_movies['genres']=='(no genres listed)']['title'].unique().tolist())/len(all_movies['title'].unique().tolist()))*100

8.194652290716334

encodage des genres

In [24]:
genres_dummies = all_movies['genres'].str.get_dummies('|')

df = pd.concat([all_movies, genres_dummies], axis=1)

In [25]:
df

movieId                    title    year  \
0               1                Toy Story  1995.0   
1               1                Toy Story  1995.0   
2               1                Toy Story  1995.0   
3               1                Toy Story  1995.0   
4               1                Toy Story  1995.0   
...           ...                      ...     ...   
25000090   209157                       We  2018.0   
25000091   209159       Window of the Soul  2001.0   
25000092   209163                Bad Poems  2018.0   
25000093   209169             A Girl Thing  2001.0   
25000094   209171  Women of Devil's Island  1962.0   

                                               genres  userId  rating  \
0         Adventure|Animation|Children|Comedy|Fantasy       2     3.5   
1         Adventure|Animation|Children|Comedy|Fantasy       3     4.0   
2         Adventure|Animation|Children|Comedy|Fantasy       4     3.0   
3         Adventure|Animation|Children|Comedy|Fantasy       5     4.0   
4         Adventure|Animation|Children|Comedy|Fantasy       8     4.0   
...                                               ...     ...     ...   
25000090                                        Drama  119571     1.5   
25000091                                  Documentary  115835     3.0   
25000092                                 Comedy|Drama    6964     4.5   
25000093                           (no genres listed)  119571     3.0   
25000094                       Action|Adventure|Drama  119571     3.0   

           timestamp  (no genres listed)  Action  Adventure  ...  Film-Noir  \
0         1141415820                   0       0          1  ...          0   
1         1439472215                   0       0          1  ...          0   
2         1573944252                   0       0          1  ...          0   
3          858625949                   0       0          1  ...          0   
4          890492517                   0       0          1  ...          0   
...              ...                 ...     ...        ...  ...        ...   
25000090  1574280748                   0       0          0  ...          0   
25000091  1574280985                   0       0          0  ...          0   
25000092  1574284913                   0       0          0  ...          0   
25000093  1574291826                   1       0          0  ...          0   
25000094  1574291937                   0       1          1  ...          0   

          Horror  IMAX  Musical  Mystery  Romance  Sci-Fi  Thriller  War  \
0              0     0        0        0        0       0         0    0   
1              0     0        0        0        0       0         0    0   
2              0     0        0        0        0       0         0    0   
3              0     0        0        0        0       0         0    0   
4              0     0        0        0        0       0         0    0   
...          ...   ...      ...      ...      ...     ...       ...  ...   
25000090       0     0        0        0        0       0         0    0   
25000091       0     0        0        0        0       0         0    0   
25000092       0     0        0        0        0       0         0    0   
25000093       0     0        0        0        0       0         0    0   
25000094       0     0        0        0        0       0         0    0   

          Western  
0               0  
1               0  
2               0  
3               0  
4               0  
...           ...  
25000090        0  
25000091        0  
25000092        0  
25000093        0  
25000094        0  

[25000095 rows x 27 columns]

To do :
- ajouter la desciption de chaque film (essayer de voir si on ne peut pas scrapper)
- studio ?
- acteurs ?